rotkehlchen/accounting/cost_basis/base.py

```diff
@@ -150,13 +158,19 @@ def __len__(self) -> int:
 class FIFOAcquisitionsOrder(BaseAcquisitionsOrder):
     """Accounting in FIFO (first-in-first-out) order"""
     def add_acquisition(self, acquisition: AssetAcquisitionEvent) -> None:
-        self._acquisitions.append(acquisition)
+        heapq.heappush(self._acquisitions, (Timestamp(time.time_ns()), acquisition))
 
 
 class LIFOAcquisitionsOrder(BaseAcquisitionsOrder):
     """Accounting in LIFO (last-in-first-out) order"""
     def add_acquisition(self, acquisition: AssetAcquisitionEvent) -> None:
-        self._acquisitions.appendleft(acquisition)
+        heapq.heappush(self._acquisitions, (Timestamp(-time.time_ns()), acquisition))
+
```

Comment:
I believe you should use the timestamp fo the event here. Doesn't make more sense? You are using ns here to avoid to having the same timestamp (correct me if wrong) and this is prone to errors